fix(docker): nightly dev-php-fpm 8.6 build broken by obsolete --with-pic flag

## Summary

The "Nightly Build development 8.6 php-fpm Docker" workflow has been failing since March 12, 2026. The PHP `./configure` step fails with:

```
configure: error: The Libtool --with-pic/--without-pic configure option is obsolete.
```

## Details

The Dockerfile at `docker/library/dockers/dev-php-fpm/Dockerfile` clones `php/php-src` HEAD (no pinned tag or branch) and runs `./buildconf --force` to regenerate the configure script. A commit landed in php-src around March 11-12 that updated the generated configure to use a newer Libtool version which rejects `--with-pic` as obsolete. Combined with `--enable-option-checking=fatal` (line 152), this turns the deprecation into a fatal error.

**Last passing run:** [March 11](https://github.com/openemr/openemr/actions/runs/22932407902)
**First failing run:** [March 12](https://github.com/openemr/openemr/actions/runs/22982673302)

## Suggested fix

1. **Remove `--with-pic`** (lines 157-158). PIC code is already generated via `-fpic` in the `PHP_CFLAGS` env var (line 55), so the flag is redundant.
2. **Pin the php-src clone to a tag or branch.** The Dockerfile currently clones HEAD of php-src master, making the build vulnerable to any upstream change. The commented-out code on lines 93-99 shows this was once pinned to a PHP 8.2 tag — a similar approach should be used for 8.6.

--- a/.github/workflows/build-dev-php-fpm-docker.yml
+++ b/.github/workflows/build-dev-php-fpm-docker.yml
@@ -3,33 +3,35 @@ name: Nightly Build development 8.6 php-fpm Docker
 on:
   workflow_dispatch:
   schedule:
-    - cron: '0 1 * * *' # (daily) run at 1 AM UTC
+  - cron: '0 1 * * *'   # (daily) run at 1 AM UTC
 
 permissions:
   contents: read
 
 jobs:
   build:
-    # Only run from master branch on the main repository
-    if: github.repository_owner == 'openemr' && github.repository == 'openemr/openemr' && github.ref == 'refs/heads/master'
     runs-on: ubuntu-24.04
+    env:
+      # Only publish the image from master of the canonical repo; other branches / forks build dry-run.
+      PUSH_IMAGE: ${{ github.repository == 'openemr/openemr' && github.ref == 'refs/heads/master' }}
     steps:
-      - name: Checkout repository
-        uses: actions/checkout@v6
-      - name: Set up QEMU
-        uses: docker/setup-qemu-action@v4
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v4
-      - name: Login to Docker Hub
-        uses: docker/login-action@v4
-        with:
-          username: ${{ secrets.DOCKERHUB_USERNAME }}
-          password: ${{ secrets.DOCKERHUB_TOKEN }}
-      - name: Build and push dev-php-fpm:pre-build-dev-86 docker
-        uses: docker/build-push-action@v7
-        with:
-          context: ./docker/library/dockers/dev-php-fpm
-          tags: openemr/dev-php-fpm:pre-build-dev-86
-          platforms: linux/amd64
-          push: true
-          no-cache: true
+    - name: Checkout repository
+      uses: actions/checkout@v6
+    - name: Set up QEMU
+      uses: docker/setup-qemu-action@v4
+    - name: Set up Docker Buildx
+      uses: docker/setup-buildx-action@v4
+    - name: Login to Docker Hub
+      if: env.PUSH_IMAGE == 'true'
+      uses: docker/login-action@v4
+      with:
+        username: ${{ secrets.DOCKERHUB_USERNAME }}
+        password: ${{ secrets.DOCKERHUB_TOKEN }}
+    - name: Build (and push on master) dev-php-fpm:pre-build-dev-86 docker
+      uses: docker/build-push-action@v7
+      with:
+        context: ./docker/library/dockers/dev-php-fpm
+        tags: openemr/dev-php-fpm:pre-build-dev-86
+        platforms: linux/amd64
+        push: ${{ env.PUSH_IMAGE == 'true' }}
+        no-cache: true

--- a/docker/library/dockers/dev-php-fpm/Dockerfile
+++ b/docker/library/dockers/dev-php-fpm/Dockerfile
@@ -90,6 +90,8 @@ RUN set -eux; \
 #	fi; \
 	git clone https://github.com/php/php-src php; \
 	cd php; \
+# PHP 8.6 only exists on master today (no 8.6 tag or PHP-8.6 branch yet), so we clone master.
+# The point of this nightly image is to catch breakage against in-development PHP 8.6 early.
 #	git checkout "$( \
 #		git for-each-ref --format='%(refname)' refs/tags \
 #		| grep -E 'refs/tags/php-8\.2[.0-9]+$' \
@@ -154,9 +156,7 @@ RUN set -eux; \
 # https://github.com/docker-library/php/issues/439
 		--with-mhash \
 		\
-# https://github.com/docker-library/php/issues/822
-		--with-pic \
-		\
+# --with-pic removed: newer Libtool rejects it as obsolete. PIC code is already produced via -fpic in PHP_CFLAGS above.
 # --enable-ftp is included here because ftp_ssl_connect() needs ftp to be compiled statically (see https://github.com/docker-library/php/issues/236)
 		--enable-ftp \
 # --enable-mbstring is included here because otherwise there's no way to get pecl to use it properly (see https://github.com/docker-library/php/issues/195)

